core/common/src/main/java/alluxio/security/authentication/ImpersonationAuthenticator.java

```diff
@@ -146,4 +123,39 @@ public void authenticate(String connectionUser, String impersonationUser)
         connectionUser, impersonationUser, connectionUser, impersonationUser,
         RuntimeConstants.ALLUXIO_SECURITY_DOCS_URL));
   }
+
+  @Override
+  public void update() {
+    loadImpersonationUser(Configuration.global());
+  }
+
+  private void loadImpersonationUser(AlluxioConfiguration conf) {
+    Map<String, Set<String>> impersonationGroups = new HashMap<>();
+    Map<String, Set<String>> impersonationUsers = new HashMap<>();
+    for (PropertyKey key : conf.keySet()) {
```

Comment:
If you use `userKeySet()` this method, it can reduce loop count significantly

Reply:
done.